src/Data/Fin/Induction.agda

```diff
@@ -59,6 +59,20 @@ open WF public using (Acc; acc)
   induct {suc i} (acc rec) = Pᵢ⇒Pᵢ₊₁ i (induct (rec (inject₁ i) i<i+1))
     where i<i+1 = ℕ<⇒inject₁< (i<1+i i)
 
+≤-Induction : {k : Fin (suc n)} (P : ∀ {p} → k ≤ p → Set ℓ)
```

Comment:
Hmm so I've been trying to work out what this is actually doing! So you start at `P k` and then prove `P (k+1)`, `P (k+2)` etc. etc. right?

It feels like parameterising the predicate `P` by the inequality is overkill. Could you reformulate the property as:

```agda
≤-Induction : (P : Pred (Fin (suc n)) ℓ) →
              ∀ {k : Fin (suc n)} → P k →
              (ps : ∀ {m} → m ≥ k → P (inject₁ m) → P (suc m)) →
              ∀ {m} → m ≥ k → P m
```
which I feel is simpler and more natural?

Reply:
I used your suggestion.